android/app/src/main/java/com/reactnativenavigation/activities/BottomTabActivity.java

```diff
@@ -104,14 +104,15 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public void push(Screen screen) {
         super.push(screen);
+        setNavigationStyle(screen);
         mScreenStacks.get(mCurrentStackPosition).push(screen);
     }
 
     @Override
     public Screen pop(String navigatorId) {
         super.pop(navigatorId);
         Screen screen = mScreenStacks.get(mCurrentStackPosition).pop();
-        setNavigationStyle(screen);
+        setNavigationStyle(getCurrentScreen());
```

Comment:
When popping it was applying the style of the screen that had just been popped rather than the new screen.